### DISCUSSION TITLE
WhatsApp Cloud API

### DISCUSSION THREAD

**Discussion by saikat-here** [chatwoot/chatwoot#4710] [Feature Requests]
WhatsApp cloud API launched for sending and receiving messages.
A business owner need not to pay for hosting a number to third party.

Is there any plan to implement that ?

https://developers.facebook.com/docs/whatsapp/cloud-api

**Comment by sojan-official** [comment]
@saikat-here  Yes, we will be prioritising a native integration in upcoming releases. Feel free to track the progress over

https://github.com/chatwoot/chatwoot/issues/4712

**Comment by saikat-here** [comment]
I have upgraded yesterday to the latest version but I can see only Twilo and 360Dialog options only. 

![image](https://user-images.githubusercontent.com/2095290/179917810-2ddb26ea-7c98-46ea-a6c5-40749ad367c6.png)


Am I doing anything wrong ?

**Reply by saikat-here** [reply]
@sojan-official , could you please help on this ?

**Reply by sojan-official** [reply]
could you give details of the installation environment you are using ?

**Reply by saikat-here** [reply]
OS :  Ubuntu 20.04.4 LTS

I have used the following commands to upgrade.


sudo -i -u chatwoot
cd chatwoot
git checkout master && git pull
rvm install "ruby-3.0.4"
rvm use 3.0.4 --default
bundle
yarn
rake assets:precompile RAILS_ENV=production
RAILS_ENV=production bundle exec rake db:migrate
exit
cp /home/chatwoot/chatwoot/deployment/chatwoot-web.1.service /etc/systemd/system/chatwoot-web.1.service
cp /home/chatwoot/chatwoot/deployment/chatwoot-worker.1.service /etc/systemd/system/chatwoot-worker.1.service
cp /home/chatwoot/chatwoot/deployment/chatwoot.target /etc/systemd/system/chatwoot.target
systemctl daemon-reload
systemctl restart chatwoot.target

**Reply by sojan-official** [reply]
Could you try this and restart? 
https://www.chatwoot.com/docs/self-hosted/deployment/linux-vm#if-precompile-fails

**Reply by saikat-here** [reply]
I am getting this output.

![image](https://user-images.githubusercontent.com/2095290/180031836-9c7405ae-ad45-4d45-8881-f21b30fc2f03.png)

**Reply by saikat-here** [reply]
It worked while second time. Not sure why it failed first time. No issues now.
